Lesson_5/main.js

```diff
@@ -0,0 +1,63 @@
+// 1. Добавить методы и обработчики событий для поля поиска. Создать в объекте данных поле searchLine и привязать к нему содержимое поля ввода. На кнопку «Искать» добавить обработчик клика, вызывающий метод FilterGoods.
+// 2. Добавить корзину. В html-шаблон добавить разметку корзины. Добавить в объект данных поле isVisibleCart, управляющее видимостью корзины.
+// 3. *Добавлять в .goods-list заглушку с текстом «Нет данных» в случае, если список товаров пуст.
+
+const API = 'https://raw.githubusercontent.com/GeekBrainsTutorial/online-store-api/master/responses/';
+
+const app = new Vue({
+  el: '#app',
+  data: {
+    products: [],
+    cartProducts: [],
+    filteredProducts: [],
+    searchLine: '',
+    isVisibleCart: true
+  },
+  methods: {
+    filter() {
+      this.filteredProducts = this.searchLine
+        ? this.filteredProducts.filter(el => { return el.product_name.toLowerCase().includes(this.searchLine); })
+        : this.products;
+    },
+    add(item) {
+      this.fetch('addToBasket', data => {
+        if (data.result === 1) {
+          if (this.cartProducts.indexOf(item)) {
```

Comment:
indexOf возвращает index элемента или -1 если элемента нет
```suggestion
          if (this.cartProducts.indexOf(item) !== -1) {
```